x-pack/plugins/observability_solution/apm/server/utils/es_fields_mappings.ts

```diff
@@ -573,8 +575,27 @@ export const serviceMetadataIconsMapping = (
         name: normalizeValue<string>(fields[CLOUD_PROVIDER]),
       },
     },
-    container: normalizeValue<Container>(fields[CONTAINER]),
-    kubernetes: normalizeValue<Kubernetes>(fields[KUBERNETES]),
+    container: {
```

Comment:
One big issue that needs to be resolved here. The icons only check if `kubernetes:` is not null, so this structure will result in the kubernetes Icon always being shown. 
What would the best pattern in javascript be to conditionally add each field? something like:
```
 kubernetes: {
      pod: {
       normalizeValue<string>(fields[KUBERNETES_POD_NAME]) ? name: normalizeValue<string>(fields[KUBERNETES_POD_NAME]) : null,
...
      },
```

Reply:
@bryce-b thank you for solving it